src/store/slices/hotspotsSlice.ts

```diff
@@ -25,86 +27,81 @@ const initialState: HotspotsByCluster = {
   testnet: {},
 }
 
-export const fetchHotspots = createAsyncThunk(
-  'hotspots/fetchHotspots',
+export const fetchAllHotspots = createAsyncThunk(
+  'hotspots/fetchAllHotspots',
   async ({
     account,
     anchorProvider,
-    limit = DEFAULT_PAGE_AMOUNT,
     cluster: _cluster,
   }: {
     account: CSAccount
     anchorProvider: AnchorProvider
     cluster: Cluster
-    limit?: number
   }) => {
     if (!account.solanaAddress) throw new Error('Solana address missing')
-
     const pubKey = new PublicKey(account.solanaAddress)
-    const fetchedHotspots = await solUtils.getCompressedCollectablesByCreator(
+    let fetchedHotspots = await solUtils.getCompressedCollectablesByCreator(
       pubKey,
       anchorProvider,
       1,
-      limit,
-    )
-
-    const hotspotsWithMetadata = await solUtils.getCompressedNFTMetadata(
-      fetchedHotspots.items,
+      1000,
     )
 
-    const hotspotsWithPendingRewards =
-      await solUtils.annotateWithPendingRewards(
+    let hotspots: CompressedNFT[] = []
+    while (hotspots.length < fetchedHotspots.total) {
```

Comment:
This is the spot that's causing a second searchAssets fetch. You can use fetchedHotspots.grandTotal to figure out if you need to fetch more